### ISSUE TITLE
Command for removing expired sectors

### ISSUE THREAD

**Issue by benjaminh83** [filecoin-project/lotus#7119]
### Checklist

- [X] This is **not** a new feature or an enhancement to the Filecoin protocol. If it is,  please open an [FIP issue](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md).
- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated feature request to propose.

### Lotus component

lotus miner - proving(WindowPoSt)

### What is the motivation behind this feature request? Is your feature request related to a problem? Please describe.

More and more storage providers are going to need a tool to handle expired sectors. Lotus does not automatically flush the storage when a sector is expired (read: Is not in a proving deadline anymore). 

Currently the storage provider has to go through the sector list, look for sectors where the Expiration column is "xxx ago", double check that these sectors are not in any deadline, and then manually punch them out by:   
` lotus-miner sectors remove --really-do-it xxx`
 
If storage provider puts in any wrong sector numbers, these sectors are gone, and only termination and slashing is waiting. This is not good!

### Describe the solution you'd like

Basically we just need a simple command that can flush out any expired sectors. I would imagine like: 
`lotus-miner sectors remove --expired`
It wouldn't need the `--really-do-it` as lotus would make sure only sectors that is not in any proving deadline is deleted, and therefore this command **should be totally safe**!

**Comment by funktafide** [comment]
This would be a great command to have.  I have some of those sectors sitting in my list for sure. Thanks for submitting @benjaminh83 !

**Comment by Meatball13** [comment]
Yes please...right now it's a manual process to go through and do it and on more than one occasion I've accidentally removed good sectors.  If there was a command that would just go through, clean out anything that's expired or is no longer 'live' (say a CC sector that was upgraded, that would be great.  

Maybe one suggestion though, if you run the command, show a list of everything it's about to nuke and then give the person an opportunity to yes or no it.

**Comment by neondragon** [comment]
Related: Terminated sectors are also hard to locate and remove.

**Comment by f8-ptrk** [comment]
if this could get rid of unsealed copies of expired deals too - even better!

**Comment by benjaminh83** [comment]
@f8-ptrk Yes, this should also delete the unsealed sector for fast retrieval. We don't want the storage providers to be manually cleaning up the unsealed. Although they are not "proved", its still very easy to delete the wrong files, and the is bad for storage provider and for the client.

**Comment by f8-ptrk** [comment]
@benjaminh83 even if the sector is extended the unsealed copy could be removed

**Comment by benjaminh83** [comment]
@f8-ptrk It seems like the unsealed copy is also deleted, using the remove command. I just tried one expired sector, and my miner logs show: 

```
2021-08-18T14:39:06.432+0200	INFO	stores	stores/local.go:616	remove /mnt/md121/storage/sealed/s-t02576-586
2021-08-18T14:39:08.794+0200	INFO	stores	stores/local.go:616	remove /mnt/md121/storage/cache/s-t02576-586
2021-08-18T14:39:08.880+0200	INFO	stores	stores/local.go:616	remove /mnt/md121/storage/unsealed/s-t02576-586
```

**Comment by s0nik42** [comment]
cannot wait for it !

**Comment by benjaminh83** [comment]
We need it @jennijuju

**Comment by jennijuju** [comment]
Closed by https://github.com/filecoin-project/lotus/commit/927ef041f8d86c8dd86798f35988278b3cfa1efe#
